DRIQuery.cpp

```diff
@@ -10,44 +10,113 @@
 
 
 DRIQuery::DRIQuery() {
+
+}
+
+
+bool DRIQuery::isSystemSupported() {
+    if (std::string(std::getenv("XDG_SESSION_TYPE")) == "wayland") {
+#ifdef WAYLAND_PROTOCOL
+        HelpersWayland hw;
+        return hw.hasProperLibEGL();
+#endif
+    } else if (std::string(std::getenv("XDG_SESSION_TYPE")) == "x11") {
+        this->getScreenDriver = (glXGetScreenDriver_t *) glXGetProcAddress((const GLubyte *) "glXGetScreenDriver");
+        this->getDriverConfig = (glXGetDriverConfig_t *) glXGetProcAddress((const GLubyte *) "glXGetDriverConfig");
+        this->getGlxExtensionsString = (glXQueryExtensionsString_t *) glXGetProcAddress((const GLubyte *) "glXQueryExtensionsString");
+
+        if (!this->getScreenDriver || !this->getDriverConfig || !this->getGlxExtensionsString) {
+            std::cerr << _("Error getting function pointers. LibGL must be too old.") << std::endl;
+            return false;
+        } else {return true;}
+    } else {
+        return false;
+    }
+
+    return false;
+}
+
+const char *DRIQuery::queryDN(int s) {
+    Display *display;
+    const char *ret;
+
+    if (!(display = XOpenDisplay(nullptr))) {
+        std::cerr << _("Couldn't open X display") << std::endl;
+        return "";
+    }
+
     this->getScreenDriver = (glXGetScreenDriver_t *) glXGetProcAddress((const GLubyte *) "glXGetScreenDriver");
-    this->getDriverConfig = (glXGetDriverConfig_t *) glXGetProcAddress((const GLubyte *) "glXGetDriverConfig");
-    this->getGlxExtensionsString = (glXQueryExtensionsString_t *) glXGetProcAddress((const GLubyte *) "glXQueryExtensionsString");
+    ret = (*(this->getScreenDriver))(display, s);
 
-    if (!this->getScreenDriver || !this->getDriverConfig || !this->getGlxExtensionsString) {
-        std::cerr << _("Error getting function pointers. LibGL must be too old.") << std::endl;
+    XCloseDisplay(display);
+
+    return ret;
+}
+
+const char *DRIQuery::queryDC(const char *dn) {
```

Comment:
What is `DC`?
What is `dn`?
Please give more descriptive names to the variables.